api/konnect/v1alpha1/konnect_extension_types.go

```diff
@@ -0,0 +1,167 @@
+	// +kubebuilder:validation:XValidation:rule="self.provisioning == 'Automatic' ? !has(self.secretRef) : true",message="secretRef must not be set when provisioning is set to Automatic."
+	// +kubebuilder:validation:Required
+	CertificateSecret *CertificateSecret `json:"certificateSecret,omitempty"`
+}
+
+const (
+	// ManualSecretProvisioning is the method used to provision the certificate manually.
+	ManualSecretProvisioning = "Manual"
+	// AutomaticSecretProvisioning is the method used to provision the certificate automatically.
+	AutomaticSecretProvisioning = "Automatic"
+)
+
+// CertificateSecret contains the information to access the client certificate.
+type CertificateSecret struct {
+	// Provisioning is the method used to provision the certificate. It can be either Manual or Automatic.
+	// In case manual provisioning is used, the certificate must be provided by the user.
+	// In case automatic provisioning is used, the certificate will be automatically generated by the system.
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum=Manual;Automatic
+	// +kubebuilder:default=Automatic
+	Provisioning *string `json:"provisioning,omitempty"`
+
+	// CertificateSecretRef is the reference to the Secret containing the client certificate.
+	// +kubebuilder:validation:Optional
+	CertificateSecretRef *SecretRef `json:"secretRef,omitempty"`
+}
+
+// SecretRef contains the reference to the Secret containing the Konnect Control Plane's cluster certificate.
+type SecretRef struct {
+	// Name is the name of the Secret containing the Konnect Control Plane's cluster certificate.
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+}
+
+// KonnectExtensionStatus defines the observed state of KonnectExtension.
+type KonnectExtensionStatus struct {
+	// DataPlaneRefs is the array  of DataPlane references this is associated with.
+	// A new reference is set by the operator when this extension is associated with
+	// a DataPlane through its extensions spec.
+	//
+	// +kubebuilder:validation:MaxItems=16
+	DataPlaneRefs []commonv1alpha1.NamespacedRef `json:"dataPlaneRefs,omitempty"`
+
+	// ControlPlaneRefs is the array  of ControlPlane references this is associated with.
+	// A new reference is set by the operator when this extension is associated with
+	// a ControlPlane through its extensions spec.
+	//
+	// +kubebuilder:validation:MaxItems=16
+	ControlPlaneRefs []commonv1alpha1.NamespacedRef `json:"controlPlaneRefs,omitempty"`
+
+	// DataPlaneClientAuth contains the configuration for the client certificate authentication for the DataPlane.
+	// +kubebuilder:validation:Optional
+	DataPlaneClientAuth *DataPlaneClientAuthStatus `json:"dataPlaneClientAuth,omitempty"`
+
+	// Conditions describe the current conditions of the KonnectExtensionStatus.
+	//
+	// Known condition types are:
+	//
+	// * TODO: define condition types https://github.com/Kong/kubernetes-configuration/issues/292
+	//
```

Comment:
1. This will end up in the CRD. Can we move the comment above this field's godoc to prevent this?

```suggestion
	// TODO: define condition types https://github.com/Kong/kubernetes-configuration/issues/292
	
	// Conditions describe the current conditions of the KonnectExtensionStatus.
	//
	// Known condition types are:
	//
```

1. "Known condition types are:" Can we either list the known conditions here or drop this?